[Docs] Jemalloc troubleshooting. (#2998)
ChainSafe/forest: documentation/src/trouble_shooting.md

the size of these files lets RocksDB use less memory but runs the risk of
hitting the open-files limit. If you do hit this limit, either increase the file
size or use `ulimit` to increase the open-files limit.

#### Jemalloc issues on Apple Silicon macs

Forest is compiled with `jemalloc` as a default allocator. If you are having

> problems running it, perhaps this checklist would help:

Reviewer: ```suggestion
problems running or compiling Forest, use this checklist:
```

> 1. Make sure you are using an arm64 version of homebrew, this could be a problem

Reviewer: ```suggestion
1. Make sure you are using an arm64 version of homebrew; this could be a problem
```

   one inherits when migrating from an Intel Mac to Apple Silicon:
   [Stackoverflow example](https://stackoverflow.com/a/68443301).
2. Make sure your default host is set to `aarch64-apple-darwin` via
   `rustup set default-host aarch64-apple-darwin`.

> 3. This could result in various errors related to the fact that you still have

Reviewer: I wonder if we can make this third step easier to perform. How does one check for Intel flavoured files as compared to aarch64 files? Delete everything in `/usr/local/lib` and re-install?

Author: This is a good question. Perhaps this would work, though afaik Apple Silicon version of homebrew installs all the stuff to `/opt/homebrew/lib`.

Author: Yes, there is a better way. Fixed now.
